plugins/uber-jar/src/UberJar.scala

```
@@ -0,0 +1,136 @@
+                    compilerTarget: File,
+                    classpath: ClassPath,
+                    mainClass: Option[String],
+                    jarName: String
+                   )(log: String => Unit): Try[Unit] = {
+    val targetPath = targetDir.toPath
+    log(s"Target directory is: $targetPath")
+    log(s"Compiler targer directory is: $compilerTarget")
+    log(s"Classpath is: $classpath")
+    mainClass foreach (c => log(s"Main class is is: $c"))
+
+    val jarPath = {
+      log("Creating jar file...")
+      val validJarName = if (jarName.endsWith("*.jar")) jarName else jarName + ".jar"
+      log(s"Jar name is: $validJarName")
+      val path = targetPath.resolve(validJarName)
+      Files.deleteIfExists(path)
+      Files.createFile(path)
+      log("Creating jar file - DONE")
+      path
+    }
+    withCloseable(new JarOutputStream(new FileOutputStream(jarPath.toFile), createManifest(mainClass))) { out =>
+      writeTarget(compilerTarget, out)(log)
+
+      // it will contain all jar files, including jars, that cbt depend on! Not good!
+      val jars = classpath.files filter (f => jarFileMatcher.matches(f.toPath))
+      log(s"Found ${jars.length} jar dependencies: \n ${jars mkString "\n"}")
+      writeExtractedJars(jars, targetPath, out)(log)
+
+      // TODO: make it in try-catch-finally style
+      out.close()
+      System.err.println(s"Uber jar created. You can grab it at $jarPath")
+    }
+  }
+
+  /**
+    * Writes three attributes to manifest file:
+    *
+    *  Main-Class: classname
+    *  Manifest-Version: 1.0
+    *  Created-By: java.runtime.version
+    *
+    * @param mainClass optional main class
+    * @return mainifest for jar
+    */
+  private def createManifest(mainClass: Option[String]): jar.Manifest = {
+    val m = new jar.Manifest()
+    m.getMainAttributes.putValue("Manifest-Version", "1.0")
+    val createdBy = Option(System.getProperty("java.runtime.version")) getOrElse "1.7.0_06 (Oracle Corporation)"
```

should we fail rather than using a default value here? you could simply use `sys.props("java.runtime.version")`